docs: add few examples settings for new csi management (#17599)
rook/rook: Documentation/Helm-Charts/csi-drivers-chart.md

To configure the Ceph-CSI drivers, Rook requires the installation of the [Ceph-CSI Driver chart](https://github.com/ceph/ceph-csi-operator/blob/main/docs/helm-charts/drivers-chart.md). This chart configures the CSI drivers to provision and mount volumes to make available the Ceph storage to your applications.

> CSI deployment is admin-managed through the ceph-csi-operator. For manifest-based examples and

Reviewer: I would suggest to delete this paragraph. They will see examples in the section below. And I don't think we should link to the manifest examples here. This page should focus on the helm values.

helm repo add ceph-csi-operator https://ceph.github.io/ceph-csi-operator
helm install ceph-csi-drivers --namespace rook-ceph ceph-csi-operator/ceph-csi-drivers
```

## Custom settings

Below are some examples of common settings that may need to be customized in the CSI drivers chart. Create a values file with the desired settings and install with `-f values.yaml`.

### CSI-Addons sidecar

```yaml
operatorConfig:
  driverSpecDefaults:
    deployCsiAddons: true
```

See: [CSI-Addons sidecar](../Storage-Configuration/Ceph-CSI/csi-configuration.md#csi-addons-sidecar)

### Controller plugin replicas

```yaml
operatorConfig:
  driverSpecDefaults:
    controllerPlugin:
      replicas: 2
```

See: [Controller replicas and strategy](../Storage-Configuration/Ceph-CSI/csi-configuration.md#controller-replicas-and-strategy)

### CephFS client type

```yaml
drivers:
  cephfs:
    cephFsClientType: fuse
```

See: [CephFS client type (kernel vs FUSE)](../Storage-Configuration/Ceph-CSI/csi-configuration.md#cephfs-client-type-kernel-vs-fuse)

### RBD driver name prefix

```yaml
drivers:
  rbd:

>     name: <rook-ceph>.rbd.csi.ceph.com

Reviewer: I think the example is more clear without the brackets, just keep the brackets in the note
```suggestion
    name: rook-ceph.rbd.csi.ceph.com
```

```

**Note** The prefix `<rook-ceph>` should always be the rook operator namespace.

See: [Driver name / provisioner prefix](../Storage-Configuration/Ceph-CSI/csi-configuration.md#driver-name--provisioner-prefix)

### NFS CSI driver

```yaml
drivers:
  nfs:
    enabled: true
    name: rook-ceph.nfs.csi.ceph.com
```

See: [Enable NFS CSI driver](../Storage-Configuration/Ceph-CSI/csi-configuration.md#enable-nfs-csi-driver)

### kubelet path

```yaml
operatorConfig:
  driverSpecDefaults:
    nodePlugin:
      kubeletDirPath: /var/lib/kubelet
      enableSeLinuxHostMount: true
```

See: [Node plugin kubelet path and SELinux host mount](../Storage-Configuration/Ceph-CSI/csi-configuration.md#node-plugin-kubelet-path-and-selinux-host-mount)

### Custom CSI images

> See the default images in the [CephCSI values](https://github.com/rook/rook/blob/release-1.20/deploy/charts/rook-ceph/values.yaml#L97-L137)

Reviewer: How about this:
```
Custom CSI images are configured from the [rook-ceph](operator-chart.md#configuration) chart. 
See the default images in the `rook-ceph` chart [values](https://github.com/rook/rook/blob/release-1.20/deploy/charts/rook-ceph/values.yaml#L97-L137)
```

helm repo add ceph-csi-operator https://ceph.github.io/ceph-csi-operator
helm install ceph-csi-drivers --namespace rook-ceph ceph-csi-operator/ceph-csi-drivers
```

> ## Common settings examples

Reviewer: ```suggestion
## Custom settings
```

> Create a values file (for example `csi-driver-values.yaml`) and install/upgrade with:

Reviewer: The main install command is above. Let's simplify and just tell them to create a values.yaml. It's standard helm, so we don't need to repeat  the full command. Let's just show examples.
```suggestion
Below are some examples of common settings that may need to be customized in the CSI drivers chart. Create a values file with the desired settings and install with `-f values.yaml`.
```

```console

> helm upgrade --install ceph-csi-drivers --namespace rook-ceph \

Reviewer: ```suggestion
```

  ceph-csi-operator/ceph-csi-drivers -f csi-driver-values.yaml
```

> ### Enable CSI-Addons sidecar

Reviewer: In the title, let's just include the subject, and not the verb of the setting
```suggestion
### CSI-Addons sidecar
```

```yaml
operatorConfig:
  driverSpecDefaults:

>     deployCsiAddons: true

Reviewer: Does `deployCsiAddons` need to be set in two places? I thought it just needed to be set in `operatorConfig` if the default for all drivers, or `drivers.rbd` if it is only specific to the rbd driver.

Author: If the user wants `csiAddons` for all the drivers, they can set it in operatorConfig, and for per-driver csiAddons, they use `drivers.<name>`.  In the example, let me explain that.

drivers:
  rbd:

>     deployCsiAddons: true

Reviewer 2: IMO we dont need to enable csi-addons at two level, one should be enough

```

See: [CSI-Addons sidecar](../Storage-Configuration/Ceph-CSI/csi-configuration.md#csi-addons-sidecar)

> ### Increase controller plugin replicas

Reviewer: ```suggestion
### Controller plugin replicas
```

```yaml
operatorConfig:
  driverSpecDefaults:
    controllerPlugin:
      replicas: 2
      deploymentStrategy:

>         type: Recreate

Reviewer 2: no need to mention about deploymentStragegy as the example is about increasing the replica count?

```

See: [Controller replicas and strategy](../Storage-Configuration/Ceph-CSI/csi-configuration.md#controller-replicas-and-strategy)

> ### Set CephFS client type to FUSE

Reviewer: ```suggestion
### CephFS client type
```

```yaml
drivers:
  cephfs:
    cephFsClientType: fuse
```

See: [CephFS client type (kernel vs FUSE)](../Storage-Configuration/Ceph-CSI/csi-configuration.md#cephfs-client-type-kernel-vs-fuse)

> ### Set custom RBD driver/provisioner name prefix

Reviewer: ```suggestion
### RBD driver name prefix
```

> ```yaml

Reviewer: The prefix should always be the namespace of the rook operator, right? Let's add a comment about that.

Author: Done

drivers:
  rbd:

>     name: my-prefix.rbd.csi.ceph.com

Reviewer: ```suggestion
    name: rook-ceph.rbd.csi.ceph.com
```

```

See: [Driver name / provisioner prefix](../Storage-Configuration/Ceph-CSI/csi-configuration.md#driver-name--provisioner-prefix)

> ### Enable NFS CSI driver

Reviewer: ```suggestion
### NFS CSI driver
```

```yaml
drivers:
  nfs:
    enabled: true
    name: rook-ceph.nfs.csi.ceph.com
```

See: [Enable NFS CSI driver](../Storage-Configuration/Ceph-CSI/csi-configuration.md#enable-nfs-csi-driver)

> ### Tune kubelet path and SELinux host mount

Reviewer: ```suggestion
### kubelet path
```

```yaml
operatorConfig:
  driverSpecDefaults:
    nodePlugin:
      kubeletDirPath: /var/lib/kubelet
      enableSeLinuxHostMount: true
```

See: [Node plugin kubelet path and SELinux host mount](../Storage-Configuration/Ceph-CSI/csi-configuration.md#node-plugin-kubelet-path-and-selinux-host-mount)

> ### Override CSI images from Rook chart

Reviewer: ```suggestion
### Custom CSI images
```

> These image settings are configured via the `rook-ceph` chart and rendered into the ImageSet

Reviewer: This example needs to be on the rook-ceph helm page, not on this drivers page.

Author: I removed this as it was already there in rook-ceph helm page

ConfigMap used by CSI drivers:

```yaml
csi:
  cephcsi:
    repository: quay.io/cephcsi/cephcsi
    tag: v3.16.2
  provisioner:
    repository: registry.k8s.io/sig-storage/csi-provisioner
    tag: v6.1.1
```

> See: [Custom container images](../Storage-Configuration/Ceph-CSI/csi-configuration.md#custom-container-images)

Reviewer: No need for this link to manifest-based examples. How about instead, let's link to the chart values where the defaults can be found? That will be a good example of the possibilities of what they can change.
```suggestion
See the default images in the [CephCSI values](https://github.com/rook/rook/blob/release-1.20/deploy/charts/rook-ceph/values.yaml#L97-L137)
```

Author: Done
